src/main/webapp/app/entities/file-upload-exercise/file-upload-exercise-update.component.html

```diff
@@ -88,14 +88,31 @@ <h2 *ngIf="!fileUploadExercise.id" id="jhi-file-upload-exercise-heading-create"
                     </select>
                 </div>
                 <div class="form-group">
-                    <label class="form-control-label" jhiTranslate="artemisApp.fileUploadExercise.filePattern" for="field_filePattern">File Pattern </label>
-                    <fa-icon
-                        [icon]="'question-circle'"
-                        class="text-secondary"
-                        placement="top"
-                        ngbTooltip="{{ 'artemisApp.fileUploadExercise.filePatternInfo' | translate }}"
-                    ></fa-icon>
-                    <input required minlength="2" type="text" class="form-control" name="filePattern" id="field_filePattern" [(ngModel)]="fileUploadExercise.filePattern" />
+                    <div>
+                        <label class="form-control-label" jhiTranslate="artemisApp.fileUploadExercise.filePattern" for="field_filePattern">File Pattern </label>
+                        <fa-icon
+                            [icon]="'question-circle'"
+                            class="text-secondary"
+                            placement="top"
+                            ngbTooltip="{{ 'artemisApp.fileUploadExercise.filePatternInfo' | translate }}"
+                        ></fa-icon>
+                    </div>
+                    <input
+                        required
+                        minlength="2"
+                        type="text"
+                        class="form-control"
+                        name="filePattern"
+                        id="field_filePattern"
+                        [pattern]="filePatternRegex"
+                        [(ngModel)]="fileUploadExercise.filePattern"
+                        #filePattern="ngModel"
+                    />
+                    <ng-container *ngFor="let e of filePattern.errors! | keyvalue | removekeys: ['required']">
+                        <div *ngIf="filePattern.invalid && (filePattern.dirty || filePattern.touched)" class="alert alert-danger">
+                            <div [jhiTranslate]="'artemisApp.fileUploadExercise.fileExtensionPattern' + '.' + e.key"></div>
```

Comment:
I think you can move this one level up and use one `div` instead of two

Reply:
True, but I will leave it like this for consistency (programming-exercise-update.component.html and course-update.component.html have similar checks for the shortname)

Reply:
I think that doesn't make sense to keep consistency here, but it will be good to have fewer nested tags in the template. If we want to do this then we should move this `ng-container` into a separate method.